packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -76,7 +82,15 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
 
         await this.paypalCommerceIntegrationService.loadPayPalSdk(methodId);
 
-        this.renderButton(methodId, paypalcommerce);
+        if (paypalcommerce.onInit && typeof paypalcommerce.onInit === 'function') {
+            paypalcommerce.onInit(() => this.renderButton(methodId, paypalcommerce));
+
+            return;
+        }
+
+        if (!paypalcommerce.shouldNotRenderOnInitialization) {
```

Comment:
It might be better to update `shouldNotRenderOnInitialization` with `shouldRenderPayPalButtonOnInitialization` and set `true` as default value.